r/R/dplyr.R

```diff
@@ -46,7 +46,12 @@ arrow_dplyr_query <- function(.data) {
       # drop_empty_groups is a logical value indicating whether to drop
       # groups formed by factor levels that don't appear in the data. It
       # should be non-null only when the data is grouped.
-      drop_empty_groups = NULL
+      drop_empty_groups = NULL,
+      # arrange_vars will be a list of expressions
+      arrange_vars = list(),
+      # arrange_desc will be a logical vector indicating the sort order for each
+      # expression in arrange_vars (FALSE for ascending, TRUE for descending)
+      arrange_desc = logical()
```

Comment:
Alternatively, `arrange_vars` could be a logical vector with names being the column names and values being the `descending` logical value.

Reply:
I thought of doing that, but we would lose support for sorting by arbitrary expressions. I experimented with some various ways to avoid having two parallel structures like this, but those all seemed messier.

Reply:
Yeah I saw further down that you were trying to support arbitrary expressions, so I guess then we have to treat these more like filter/mutate than group_by.